source/common/network/connection_impl.cc

```diff
@@ -715,6 +715,12 @@ void ConnectionImpl::onWriteReady() {
     // It is possible (though unlikely) for the connection to have already been closed during the
     // write callback. This can happen if we manage to complete the SSL handshake in the write
     // callback, raise a connected event, and close the connection.
+    ENVOY_CONN_LOG(debug, "remote close", *this);
+    if (transport_socket_->sslSyscallErrorOccurred().has_value() &&
+        transport_socket_->sslSyscallErrorOccurred().value()) {
+      setFailureReason(
```

Comment:
Instead of adding `sslSyscallErrorOccurred()` to the interface, could the specific transport socket call this when an error occurs? Adding an ssl-specific bit to the transport socket interface is not ideal.